### PR TITLE
54 user profile UI

### LINKED ISSUE #54: Add an interface to display a user's profile and data

### PR DESCRIPTION
**Pull request by clementcharmillot**
Added content for the Profile fragment.

There are setters for each element in the view that will be used to update the view from data pulled from the `UserModel` when it is ready.

Clicking on trophy opens a modal showing more details about the trophies.

image

closes #54

### CODE REVIEW

**Comment by clementcharmillot** [comment]
@HugoHof I ping you as you are in charge of the database and the data we store for an user. Do you have an idea of stored data that I could add to this profile view that I could have missed?

**Comment by HugoHof** [comment]
> @HugoHof I ping you as you are in charge of the database and the data we store for an user. Do you have an idea of stored data that I could add to this profile view that I could have missed?

Maybe you can add the age of the user, the friendlist(with the friend's username). The current distance, activity time and nb Of path goals.

You can also add the average time the runner made to run one kilometer(for ex 5,00min/km). And the number of times the user has accomplish his goals.

**Comment by clementcharmillot** [comment]
@HugoHof Thanks for the info. Added the friend list and the goals reached. I didn't add the age as it doesn't look really relevant with other statistics.

**Review by HugoHof — CHANGES_REQUESTED** [review]
This is a good UI to show the user profile. The code is very clean and there is useful helpers function in the fragments to facilitate the link between the diffrentsm data in the xml elements ids. I have just make some small comments with some suggestion. Maybe in the future you can add a method in the ProfileFragment to directly link all in the element of UserModel to the corresponding xml ids.

File: app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/ProfileFragment.kt [hunk 1]
```diff
@@ -1,10 +1,64 @@
...
 class ProfileFragment : Fragment(R.layout.fragment_profile) {
-    // TODO: add profile UI and ViewModel for this screen/fragment
-    //  Leaving Profile (with a button for example) should return to the last fragment,
-    //  usually the one select in the bottom menu
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setImageIcon(R.id.IV_Trophy, R.drawable.award)
+        setImageIcon(R.id.IV_ProfilePicture, R.drawable.profile_placholderpng)
+
```
[the remark this answers is not in the document]
Comment (clementcharmillot) [reply app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/ProfileFragment.kt:19]: I've added an action when we click on the trophy count. The idea is to redirect to the trophy fragment. We might also show the trophies in a modal. What do you think?

[the remark this answers is not in the document]
Comment (HugoHof) [reply app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/ProfileFragment.kt:20]: It's a goof idea.

**Pull request merged**